v2/client.go

```diff
@@ -224,12 +236,13 @@ func (c *Client) performRequest(req *http.Request, out interface{}) error {
 			}
 		}
 
-		return json.Unmarshal(body, out)
+		err = json.Unmarshal(body, out)
+		return err
 	}
 
 	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusCreated {
 		var apiErr APIError
-		if err = json.Unmarshal(body, &apiErr); err != nil {
+		if err := json.Unmarshal(body, &apiErr); err != nil {
```

Comment:
Nit x 2 if the above is accepted to bring it back to how it was:
```suggestion
		if err = json.Unmarshal(body, &apiErr); err != nil {
```